tools/mo/openvino/tools/mo/utils/versions_checker.py

```diff
@@ -255,25 +326,34 @@ def check_requirements(framework=None, silent=True):
                           'It is highly recommended to use TensorFlow 2.x.\n',
                           extra={'is_warning': True})
 
+    # Get requirements list
     file_name = "requirements{}.txt".format(framework_suffix)
+    requirements_list = get_requirements_list_from_file(file_name, env_setup)
 
-    # if MO was run from developer clone requirements are 4 levels out from mo.py
-    requirements_file = os.path.realpath(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, os.pardir, file_name))
-    if not Path(requirements_file).is_file():
-        # if MO was run from site-packages requirements are at the same level with mo.py file
-        # in site-packages/openvino/tools/mo/requirements.txt
-        requirements_file = os.path.realpath(os.path.join(os.path.dirname(__file__), os.pardir, file_name))
+    # Get constraints list
+    constraints_list = get_requirements_list_from_file("../constraints.txt", env_setup)
```

Comment:
Please parse requirements file if it has dependency on the constrain file. If yes, you need to get the constraint file name and also parse it.

Reply:
Removed checks of unsatisfied dependencies as discussed offline.